### PR TITLE
Update all rules to handle single-element outputs

### PR DESCRIPTION
**Pull request by brynpickering**
Following last-minute review comments missed in the merging of #16, I've moved snakemake NamedList objects from the python code. Instead it is handled where possible in the snakemake files or in the main function call within the python files.

### CODE REVIEW

File: rules/capacityfactors.smk [hunk 1]
```diff
@@ -52,7 +52,7 @@ rule time_average_capacityfactor_map:
     input:
         scripts = script_dir + "capacityfactors/averages_map.py",
         id_map = "build/capacityfactors/{technology}-ids.tif",
-        timeseries = rules.capacityfactor_timeseries.output
+        timeseries = "build/capacityfactors/{technology}-timeseries.nc"
```
Comment (timtroendle) [inline rules/capacityfactors.smk]: Why did you move away from the `rule` mechanism here? I'd say we should always use it when possible.

**Review by timtroendle — APPROVED** [review]
Looks very good to me. Only one comment.

File: rules/capacityfactors.smk (hunk 1 above)
Comment (brynpickering) [reply rules/capacityfactors.smk]: When using wildcards in rule outputs, does the wildcard propagate through to the rule being called? If so, I'll revert this and a couple of others to using the rule mechanism.

Comment (timtroendle) [reply rules/capacityfactors.smk]: I think so, yes. Otherwise this couldn't have worked before, no?

Comment (brynpickering) [reply rules/capacityfactors.smk]: fair.

**Comment by timtroendle** [comment]
Is it fine if I delete the merged branches?

**Comment by brynpickering** [comment]
of course!
